packages/server/src/core/degree_status/overflow.rs

```diff
@@ -55,38 +52,36 @@ impl<'a> DegreeStatusHandler<'a> {
             Transfer::CoursesOverflow => &mut self.courses_overflow_map,
         };
         for rule in &self.catalog.credit_overflows {
-            if rule.to == bank_name {
-                if let Some(overflow) = map.get_mut(&rule.from) {
-                    if *overflow > 0.0 {
-                        let msg = match transfer {
-                            Transfer::CreditOverflow => {
-                                if let Some(course_bank) =
-                                    self.catalog.get_course_bank_by_name(&rule.from)
-                                {
-                                    if course_bank.credit.is_some() {
-                                        Some(credit_overflow_msg(*overflow, &rule.from, &rule.to))
-                                    } else {
-                                        None
-                                    }
-                                } else {
-                                    None
-                                }
-                            }
-                            Transfer::CoursesOverflow => {
-                                Some(courses_overflow_msg(*overflow, &rule.from, &rule.to))
-                            }
-                            Transfer::MissingCredit => {
-                                Some(missing_credit_msg(*overflow, &rule.from, &rule.to))
-                            }
-                        };
-                        if let Some(msg) = msg {
-                            self.degree_status.overflow_msgs.push(msg);
-                        }
-                        sum += *overflow;
-                        *overflow = 0.0;
-                    }
+            if rule.to != bank_name {
+                continue;
+            }
+            let Some(overflow) = map.get_mut(&rule.from) else {
+                continue
+            };
+            if *overflow <= 0.0 {
+                continue;
+            }
+            let msg = match transfer {
+                Transfer::CreditOverflow => self
+                    .catalog
+                    .get_course_bank_by_name(&rule.from)
+                    .and_then(|course_bank| {
+                        course_bank
+                            .credit
+                            .map(|_| credit_overflow_msg(*overflow, &rule.from, &rule.to))
```

Comment:
In other places we used ||.
Let be consistent :)

Reply:
It's not up to consistency, it has a different semantic meaning.
|| -> receives zero arguments.
|_| -> receives 1 argument, but uses the special "_" binding to ignore it.